tests/phpunit/tests/media.php

```diff
@@ -3710,6 +3710,98 @@ function() {
 		$this->assertSame( 3, $omit_threshold );
 	}
 
+	/**
+	 * Tests that wp_get_loading_attr_default() returns the expected loading attribute value before loop.
+	 *
+	 * @ticket 58211
+	 *
+	 * @covers ::wp_get_loading_attr_default
+	 *
+	 * @dataProvider data_wp_get_loading_attr_default_before_and_no_loop
+	 *
+	 * @param string $context
+	 */
+	public function test_wp_get_loading_attr_default_before_loop( $context ) {
+		global $wp_query, $wp_the_query;
+
+		$wp_query = new WP_Query( array( 'post__in' => array( self::$post_ids['publish'] ) ) );
+		$this->reset_content_media_count();
+		$this->reset_omit_loading_attr_filter();
+
+		// Lazy if not main query.
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
+
+		// Set current query as main query.
+		$wp_the_query = $wp_query;
+
+		// For main query return false for any instance before end of get_header.
+		$this->assertFalse( wp_get_loading_attr_default( $context ) );
+		add_action(
+			'get_header',
+			function() use ( $context ) {
+				$this->assertFalse( wp_get_loading_attr_default( $context ) );
+			}
+		);
+
+		do_action( 'get_header' );
+
+		// A test for before the loop but after get_header to check it returns the default, lazy-loading.
+		$this->assertSame( 'lazy', wp_get_loading_attr_default( $context ) );
```

Comment:
See my comment further above, this is different from how it should behave: Images after `get_header` but before the loop should _not_ be lazy-loaded.